src/util/fs.js

```diff
@@ -880,3 +890,30 @@ export async function readFirstAvailableStream(
 
   return {stream, triedPaths};
 }
+
+export async function getFirstWriteableFolder(paths: Iterable<string>): Promise<FolderQueryResult> {
+  const result: FolderQueryResult = {
+    skipped: [],
+    folder: null,
+  };
```

Comment:
Why don't we just return a raw string? The list of skipped directories can be deduced relatively easily by the caller anyway.

Reply:
Because I also attach the error, in case that matters. May be it is EACCESS, may be it is EROFS or may be it is something else?

Also deducing the list would require another traversal of the entire list until the selected folder is found which is a bit unnecesary.